Shell/main.cpp

```diff
@@ -70,7 +70,13 @@ static int sh_export(int argc, char** argv)
         fprintf(stderr, "usage: export variable=value\n");
         return 1;
     }
-    putenv(const_cast<char*>(String::format("%s=%s", parts[0].characters(), parts[1].characters()).characters()));
+
+    // FIXME: Yes, this leaks.
+    // Maybe LibCore should grow a CEnvironment which is secretly a map to char*,
+    // so it can keep track of the environment pointers as needed?
+    const auto& s = String::format("%s=%s", parts[0].characters(), parts[1].characters());
+    char *ev = strndup(s.characters(), s.length());
+    putenv(ev);
```

Comment:
This makes me wish we had setenv instead :)

Reply:
That one is interesting. I'm not sure how to really manage implementing things that we are supposed to delete as well as ones we are supposed to leave alone... Guess we can figure it out when the need arises :)